cocos/3d/CCTerrain.cpp

```diff
@@ -159,11 +159,9 @@ void Terrain::onDraw(const Mat4 &transform, uint32_t flags)
 
     if(_isCameraViewChanged )
     {
-        auto camPos = camera->getPosition3D();
-        auto camModelMat = camera->getNodeToWorldTransform();
-        camModelMat.transformPoint(&camPos);
+        auto m = camera->getWorldToNodeTransform();
```

Comment:
Sorry about that. In the test case, the negative camera position didn't make difference about distance caculating, so i didn't notice it.